contracts/hydro/src/contract.rs

```diff
@@ -1770,6 +1863,70 @@ fn update_voting_power_on_proposals(
     Ok(())
 }
 
+// This function will lookup the vote that needs to be updated when user locks
+// more tokens or refreshes the existing lockup. Whether some vote should be
+// updated or not is determined by the following logic:
+//
+// If user is refreshing a lock, we check if this lock was already voted with.
+// If yes, this vote will be updated. If no, we will not add a vote for it. Instead,
+// we leave user a choice to vote later.
+// Is user is adding a new lock, we check if the user already voted for some proposals.
+// If user voted for only one proposal, then the new lock power will be added to it.
+// If user voted for multiple proposals, then we do not add a vote for new lock entry.
+fn get_vote_for_update(
```

Comment:
this could probably use its own unit test; the logic above is already a pretty good guide as to what to test

Reply:
Done in https://github.com/informalsystems/hydro/pull/157/commits/5158433fc44926ddc3da7c85ce47c05b7c15ccbe